src/darwin/Framework/CHIP/MTROTAUnsolicitedBDXMessageHandler.mm

```diff
@@ -0,0 +1,145 @@
+    // TODO: #36181 - This class needs to keep a list of all MTROTAImageTransferHandlers mapped to the fabric index and only
+    // delete the MTROTAImageTransferHandler objects with a fabric index matching the MTRDeviceController's fabric index.
+    Shutdown();
+}
+
+CHIP_ERROR MTROTAUnsolicitedBDXMessageHandler::OnUnsolicitedMessageReceived(const PayloadHeader & payloadHeader, const SessionHandle & session,
+    ExchangeDelegate * _Nonnull & newDelegate)
+{
+    assertChipStackLockedByCurrentThread();
+
+    ChipLogDetail(BDX, "MTROTAUnsolicitedBDXMessageHandler: OnUnsolicitedMessageReceived: message " ChipLogFormatMessageType " protocol " ChipLogFormatProtocolId,
+        payloadHeader.GetMessageType(), ChipLogValueProtocolId(payloadHeader.GetProtocolID()));
+
+    VerifyOrReturnError(mExchangeMgr != nullptr, CHIP_ERROR_INCORRECT_STATE);
+
+    if (GetNumberOfDelegates() >= 1) {
+        return CHIP_ERROR_BUSY;
+    }
+
+    // Only proceed if there is a valid fabric index for the SessionHandle.
+    if (session->IsSecureSession() && session->AsSecureSession() != nullptr && session->AsSecureSession()->GetFabricIndex() != kUndefinedFabricIndex) {
+
+        // If we receive a ReceiveInit BDX message, create a new MTROTAImageTransferHandler and register it
+        // as the handler for all BDX messages that will come over this exchange.
+        if (payloadHeader.HasMessageType(MessageType::ReceiveInit)) {
+            mOTAImageTransferHandler = new MTROTAImageTransferHandler();
+            newDelegate = mOTAImageTransferHandler;
+            return CHIP_NO_ERROR;
+        }
+    }
+    return CHIP_ERROR_INCORRECT_STATE;
+}
+
+void MTROTAUnsolicitedBDXMessageHandler::OnExchangeCreationFailed(ExchangeDelegate * delegate)
+{
+    assertChipStackLockedByCurrentThread();
+    auto * otaTransferHandler = static_cast<MTROTAImageTransferHandler *>(delegate);
+    VerifyOrReturn(otaTransferHandler != nullptr);
+
+    otaTransferHandler->DestroySelf();
+}
+
+uint8_t MTROTAUnsolicitedBDXMessageHandler::GetNumberOfDelegates()
+{
+    assertChipStackLockedByCurrentThread();
+    return MTROTAUnsolicitedBDXMessageHandler::mNumberOfDelegates;
+}
+
+void MTROTAUnsolicitedBDXMessageHandler::IncrementNumberOfDelegates()
+{
+    assertChipStackLockedByCurrentThread();
+    MTROTAUnsolicitedBDXMessageHandler::mNumberOfDelegates++;
+}
+
+void MTROTAUnsolicitedBDXMessageHandler::DecrementNumberOfDelegates()
+{
+    assertChipStackLockedByCurrentThread();
+    MTROTAUnsolicitedBDXMessageHandler::mNumberOfDelegates--;
+}
+
+void MTROTAUnsolicitedBDXMessageHandler::OnDelegateCreated(void * imageTransferHandler)
+{
+    assertChipStackLockedByCurrentThread();
+
+    // TODO: #36181 - Store the imageTransferHandler in a set of MTROTAImageTransferHandler objects.
+    mOTAImageTransferHandler = static_cast<MTROTAImageTransferHandler *>(imageTransferHandler);
```

Comment:
@bzbarsky-apple should i re-assign this here or just verify that it matches. if it doesn't clean up everything by calling destroyself?